lib/models/room-state.js

```diff
@@ -225,6 +225,52 @@ RoomState.prototype.getUserIdsWithDisplayName = function(displayName) {
 };
 
 /**
+ * Returns true if the given user ID has permission to send a state
+ * event of type `stateEventType` into this room.
+ * @param {string} type The type of state events to test
+ * @param {string} userId The user ID of the user to test permission for
+ * @return {boolean} true if the given user ID should be permitted to send
+ *                        the given type of state event into this room,
+ *                        according to the room's state.
+ */
+RoomState.prototype.maySendStateEvent = function(stateEventType, userId) {
+    if (!this.getMember(userId)) { return false; }
```

Comment:
You can also help yourself by using `RoomMember.powerLevel` to get the actual power level of that member, which already takes into account `users_default` and explicit `users` entries.